### PR TITLE
BE: Picklist relations issues

### PR DESCRIPTION
**Pull request by davide-alghi**
1. **Module selector is not select2 rendered** after change because
https://github.com/YetiForceCompany/YetiForceCRM/blob/ec513695023b58e7491cc465a16929f2a9ae1365/public_html/layouts/basic/modules/Settings/PickListDependency/resources/PickListDependency.js#L471
calls ```changeSelectElementView()``` with ```select``` element: it needs container

2. **'All' module selector item issues an js error** because an empty value is sent and got it by ```$request->getByType``` in 
https://github.com/YetiForceCompany/YetiForceCRM/blob/ec513695023b58e7491cc465a16929f2a9ae1365/modules/Settings/PickListDependency/views/List.php#L16
and
https://github.com/YetiForceCompany/YetiForceCRM/blob/ec513695023b58e7491cc465a16929f2a9ae1365/modules/Settings/PickListDependency/views/List.php#L27
it fails on ```purifiedValuesByType``` array with empty index in
https://github.com/YetiForceCompany/YetiForceCRM/blob/ec513695023b58e7491cc465a16929f2a9ae1365/app/Request.php#L156

picklist-relations

### CODE REVIEW

File: modules/Settings/PickListDependency/views/List.php [hunk 1]
```diff
@@ -13,7 +13,7 @@ class Settings_PickListDependency_List_View extends Settings_Vtiger_List_View
 	public function preProcess(\App\Request $request, $display = true)
 	{
 		$moduleModelList = Settings_PickListDependency_Module_Model::getPicklistSupportedModules();
-		$forModule = $request->getByType('formodule', 'Alnum');
+		$forModule = $request->get('formodule');
```
Comment (TomaszQr) [inline modules/Settings/PickListDependency/views/List.php]: We recomend use function isEmpty

$forModule = '';
if (!$request->isEmpty('formodule')) {
			$forModule = $request->getByType('formodule', 'Alnum');
}

File: modules/Settings/PickListDependency/views/List.php [hunk 2]
```diff
@@ -24,12 +24,10 @@ public function process(\App\Request $request)
 	{
 		if ($request->isAjax()) {
 			$moduleModelList = Settings_PickListDependency_Module_Model::getPicklistSupportedModules();
-			$forModule = $request->getByType('formodule', 'Alnum');
-
+			$forModule = $request->get('formodule');
```
Comment (TomaszQr) [inline modules/Settings/PickListDependency/views/List.php]: We recomend use function isEmpty

$forModule = '';
if (!$request->isEmpty('formodule')) {
			$forModule = $request->getByType('formodule', 'Alnum');
}

File: modules/Settings/PickListDependency/views/List.php (hunk 1 above)
Comment (davide-alghi) [reply modules/Settings/PickListDependency/views/List.php]: done

File: modules/Settings/PickListDependency/views/List.php (hunk 2 above)
Comment (davide-alghi) [reply modules/Settings/PickListDependency/views/List.php]: done

**Pull request merged**